client/blocks/site/index.jsx

```diff
@@ -145,3 +147,10 @@ export default React.createClass( {
 		);
 	}
 } );
+
+export default connect( ( state, { site } ) => {
+	if ( isNumber( site ) ) { // Accept site IDs
+		return { site: getSite( state, site ) };
+	}
+	return { site };
```

Comment:
Note that since we're also overriding site, we'll only be fetching the site once, which may leave a stale site in this component.